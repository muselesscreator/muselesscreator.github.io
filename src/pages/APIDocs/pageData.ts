type APIProject = { description: string; title: string; url: string };
export const projects = [
  {
    description: 'A shallow snapshot testing utility for React',
    title: 'react-shallow-snapshot',
    url: 'https://musely.github.io/react-shallow-snapshot/',
  },
  {
    description: 'A key-safe object wrapper for JS',
    title: 'StrictDict',
    url: 'https://musely.github.io/strict-dict/',
  },
  {
    description: 'A more testable React hook for keyed state management',
    title: 'useKeyedState',
    url: 'https://musely.github.io/use-keyed-state/',
  },
] as APIProject[];
export default projects;
