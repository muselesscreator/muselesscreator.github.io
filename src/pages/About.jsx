import 'react';

const About = () => {
  return (
    <article className="about active">
      <header>
        <h2 className="h2 article-title">About me</h2>
      </header>
      <div className="about-text">
        <p>With over a decade of software engineering, I specialize in helping teams produce solid, testable frontend-code and maintainable agile frontend development processes, as well as experience in building and managing communities devoted to frontend communication, coordination, and training.</p>
        <p>I have experience guiding large projects from conception, through ideation and refinement, implementation, validation, release, and maintenance, as well as coming into existing projects and codebases and updating them to a consistent level of quality.</p>
        <p>My background in SQA and the odd variety of engineering tech my career has exposed me to give me a mindset and focus on maintainability of code and practices above all else.  I like to spend a bit of thoughtful effort now to make less work for everyone later.</p>
        <p>My core goal in any code-base is not only to achieve excellent code, but specifically to make achieving excellence as easy as possible.  I have high standards for my code and the projects I own, but I also endeavor to provide tools, workflows, and utilities to allow developers to quickly build well-designed, easily-maintained, and highly/deeply tested code.</p>
        <p>It has also been my experience that Agile workflows and processes are very effective for managing the engineering process, and have been pleased to be able to lead and teach multiple teams in development and refinement of effective Agile practices, with a focus on iteration, clarity of ceremonies, and focus/efficacy of meetings.</p>
        <ul>
          <li>Senior Frontend/Full-Stack Software Engineer with 8+ years of experience with Python and Full-Stack Python/JavaScript Web development, with a focus on user interfaces, APIs, and testing methodologies.</li>
          <li>Experience developing in a variety of architectures, including web, mobile, asynchronous server applications, and  2D/3D gaming.</li>
          <li>Experience self-managing and leading on projects with complex requirements across departments and generating well defined, fully-featured, well-tested code.</li>
          <li>Experience with test automation, database design, and API design.</li>
        </ul>
      </div>
    </article>
  );
};

export default About;
