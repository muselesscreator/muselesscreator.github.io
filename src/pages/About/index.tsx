import pageData, { PageField } from './pageData';


export const mapFields = (d: PageField, i: number) => {
  const { type } = d;
  if (type === 'text') {
    return (<p key={i}>{d.text}</p>);
  }
  if (type === 'list') {
    return (
      <ul key={i}>
        {d.items.map((l, j) => <li key={j}>{l}</li>)}
      </ul>
    );
  }
};

const About = () => (
  <article className="about active">
    <header>
      <h2 className="h2 article-title">{pageData.title}</h2>
    </header>
    <div className="about-text">
      {pageData.fields.map(mapFields)}
    </div>
  </article>
);

export default About;
