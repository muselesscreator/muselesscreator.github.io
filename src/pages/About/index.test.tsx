import {
  vi,
  describe,
  test,
  expect,
} from 'vitest';

import { shallow } from '@muselesscreator/react-shallow-snapshot';

import About from './index';
import pageData, { StringField, ListField } from './pageData';

vi.mock('./pageData', () => ({
  ...vi.importActual('./pageData'),
  default: {
    title: 'Test Title',
    fields: [],
  },
}));

pageData.fields = [
  { type: 'text', text: 'test text 1' },
  { type: 'text', text: 'test text 2' },
  { type: 'text', text: 'test text 3' },
  { type: 'list', items: ['test item 1', 'test item 2', 'test item 3'] },
];
const { fields } = pageData;

describe('About page', () => {
  test('snapshot', () => {
    const el = shallow(<About />);
    expect(el.snapshot).toMatchSnapshot();
    const listItems = (fields[3] as ListField).items;
    expect(el.instance.matches(shallow(
      <article className="about-text">
        <p>{(fields[0] as StringField).text}</p>
        <p>{(fields[1] as StringField).text}</p>
        <p>{(fields[2] as StringField).text}</p>
        <ul>
          {listItems.map((item) => (
            <li>{item}</li>
          ))}
        </ul>
      </article>
    )));

  });
});
