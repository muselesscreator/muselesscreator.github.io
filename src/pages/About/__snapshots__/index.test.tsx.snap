// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`About page > snapshot 1`] = `
"<article
  className="about active"
>
  <header>
    <h2
      className="h2 article-title"
    >
      Test Title
    </h2>
  </header>
  <div
    className="about-text"
  >
    <p>
      test text 1
    </p>
    <p>
      test text 2
    </p>
    <p>
      test text 3
    </p>
    <ul>
      <li>
        test item 1
      </li>
      <li>
        test item 2
      </li>
      <li>
        test item 3
      </li>
    </ul>
  </div>
</article>"
`;
